### PR TITLE
Dicts integration

### PR DESCRIPTION
**Pull request by 0x1b2c**
echidna-dicts integration without worker, the worker will be integrated in streaming-integration branch.

When looking at the models of dicts, I feel redis still the better choice for them, the `set` data structure of redis perfectly meet the needs. And since these models are quit independent, I think it's OK to keep them use redis.

Also I places these modes under `app/models/dicts` directory since they are quit independent, but didn't give them a namespace for now since it require much more code changes.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,6 +14,12 @@ gem 'fazscore', github: 'transist/fazscore'
 gem 'sidekiq'
 gem 'sinatra', '>= 1.3.0', require: nil
 
+gem 'hiredis'
+gem 'redis-namespace'
+gem 'rmmseg-cpp', require: 'rmmseg'
+gem 'yajl-ruby', require: 'yajl'
+gem 'msworddoc-extractor', require: false
```
Comment (flyerhzm) [inline Gemfile:27]: yajl-ruby is not necessary here

File: config/initializers/redis.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ENV['ECHIDNA_REDIS_NAMESPACE'] = "e:#{ENV['USER']}:#{Rails.env[0]}"
+
+$redis = Redis::Namespace.new(
+  ENV['ECHIDNA_REDIS_NAMESPACE'],
+  redis: Redis.new(
+    host: ENV['ECHIDNA_REDIS_HOST'],
+    port: ENV['ECHIDNA_REDIS_PORT'],
+    driver: :hiredis
+  )
+)
```
Comment (flyerhzm) [inline config/initializers/redis.rb:8]: I prefer adding `config/redis.yml`, and read config file instead of reading ENV

File: Gemfile (hunk 1 above)
Comment (0x1b2c) [inline Gemfile:27]: Yeah, but yajl-ruby improve JSON processing performance anyway so I added it.

File: config/initializers/redis.rb (hunk 2 above)
Comment (0x1b2c) [inline config/initializers/redis.rb:8]: Both are OK for me. To keep use environment variable we can stil reuse the existing `/etc/systemd/conf.d/echidna` config file, and also we can add these config keys to `config/application.yml`.

Comment (flyerhzm) [inline config/initializers/redis.rb:8]: We don't need to use different redis namespace for different users, as a rails project, we should add config keys in a yaml file.

Comment (0x1b2c) [inline config/initializers/redis.rb:8]: I think we still need namespace for project name and different environments(d, t, p), right?

Comment (flyerhzm) [inline config/initializers/redis.rb:8]: yes, namespace could be 'echidna/production' or 'echidna/p'

**Pull request merged**